File: typescript/src/providers/bedrock.ts
// Drop-in replacement for `@aws-sdk/client-bedrock-runtime` (Converse API).
// Swap only the import specifier:
//   - import { BedrockRuntime } from "@aws-sdk/client-bedrock-runtime";
//   + import { BedrockRuntime } from "@spaturzu/sdk/bedrock";
//
// NOTE: this targets `BedrockRuntime` — the aggregated client that exposes the
// named methods `converse` / `converseStream` — NOT the low-level
// `BedrockRuntimeClient` (which has no such methods; it uses the Command form
// `client.send(new ConverseCommand(...))`, which `wrapBedrock` does not
// instrument). Use this drop-in only if your code calls `client.converse(...)`.
import {
  BedrockRuntime as RealBedrock,
  type BedrockRuntimeClientConfig,
} from "@aws-sdk/client-bedrock-runtime";
import { getDefaultSpaturzu } from "../default.js";
import type { SpaturzuClientOptions } from "./openai.js";
import type { TagInput } from "../context.js";

class BedrockRuntime extends RealBedrock {
  /** Tag each call with an agent (supplied at runtime by the wrap proxy;
   *  declared here so it's visible to TypeScript on the drop-in client). */
  declare withAgent: (agent: string, opts?: { tags?: TagInput }) => this;

  constructor(
    config: BedrockRuntimeClientConfig,
    sp: SpaturzuClientOptions = {},
  ) {
    super(config);
    return getDefaultSpaturzu().wrapBedrock(this, sp) as BedrockRuntime;
  }
}

export { BedrockRuntime };

File: python/src/spaturzu/bedrock.py
"""Drop-in replacement for boto3's bedrock-runtime client. Swap::

    - import boto3
    - client = boto3.client("bedrock-runtime", region_name="us-east-1")
    + from spaturzu.bedrock import BedrockRuntime
    + client = BedrockRuntime(region_name="us-east-1")

Targets the Converse API (client.converse / converse_stream); v1 is sync-only.
"""
from __future__ import annotations
from typing import Any, Optional

import boto3
from . import get_default_spaturzu


def BedrockRuntime(*args: Any, spaturzu: Optional[dict] = None, **kwargs: Any) -> Any:
    return get_default_spaturzu().wrap_bedrock(
        boto3.client("bedrock-runtime", *args, **kwargs), **(spaturzu or {})
    )
